testing/asserts.ts

```diff
@@ -864,3 +866,58 @@ export function unimplemented(msg?: string): never {
 export function unreachable(): never {
   throw new AssertionError("unreachable");
 }
+
+let snapshotFile: Record<string, string> | undefined = undefined;
+const updatedSnapshotFile: Record<string, unknown> = {};
+const snapshotMap: Record<string, number> = {};
+
+export async function assertSnapshot(context: Deno.TestContext, actual: unknown) {
+  const name = getName(context);
+  const count = getCount();
+  const testName = `${name} #${count}`;
+  const isUpdate = Deno.args.includes('--update');
+  if (!snapshotFile) {
+    const snapshotPath = getSnapshotPath();
+    await ensureFile(snapshotPath);
+    const {snapshot} = await import(snapshotPath);
+    snapshotFile = snapshot;
+  }
+  const snapshot = snapshotFile?.[testName] ?? '';
+  if (isUpdate) {
+    let isEqual = true;
+    try {
+      assertEquals(_format(actual), snapshot.slice(1, -1));
+    } catch {
+      isEqual = false;
+    }
+    if (!isEqual) console.info("Snapshot updated", testName);
+    updatedSnapshotFile[testName] = actual;
+    globalThis.onunload = writeSnapshotFileSync;
+  } else {
+    assertEquals(_format(actual), snapshot.slice(1, -1));
+  }
+  function getName(context: Deno.TestContext): string {
+    if (context.parent) return `${getName(context.parent)} > ${context.name}`;
+    return context.name;
+  }
+  function getCount() {
+    const count = snapshotMap?.[name] ? snapshotMap[name] : 1;
+    snapshotMap[name] = count + 1;
+    return count;
+  }
+  function getSnapshotPath() {
+    const testFile = fromFileUrl(context.origin);
+    const parts = parse(testFile);
+    return `${join(parts.dir, parts.name)}.snap`;
+  }
+  function writeSnapshotFileSync() {
+    const snapshotPath = getSnapshotPath();
+    ensureFileSync(snapshotPath);
+    const buf = ['export const snapshot = {};\n'];
+    for (const [key, value] of Object.entries(updatedSnapshotFile)) {
+      buf.push(`\nsnapshot[\`${key}\`] = \`\n${_format(value)}\n\`;\n`);
+    }
+    Deno.writeTextFileSync(snapshotPath, buf.join(""));
+    console.log('Snapshot updated!');
```

Comment:
```suggestion
    console.log(green(bold(` > ${n} snapshots updated.`)));
```

Maybe we could mimic Jest here and log how many snapshots were updated?

Reply:
Done

Reply:
Done